### ISSUE TITLE
[Wolf3D] Freeze du jeu in game...

### ISSUE THREAD

**Issue by BenjaminSiskoo** [FCare/Kronos#995]
Le jeu fonctionne avec Kronos 1.4.5 : 
https://github.com/FCare/Kronos/commit/2107d0f7acb68d96556424c980d5ba748d57ce11

Mais il freeze avec la version 1.5.0 : 
https://github.com/FCare/Kronos/commit/d081232c243abe148e2a94e24234a8937eb90403

Un des commits doit créer ce bug

[TRCKYABiso.zip](https://github.com/FCare/Kronos/files/6114919/TRCKYABiso.zip)

**Comment by fafling** [comment]
Le freeze de Kronos se produisait aléatoirement avec les versions précédentes du jeu. Avec celle-là il se produit systématiquement dès le début du jeu en tirant en continu (bouton C maintenu appuyé).
Même comportement dans les 2 noyaux.
Pas de message dans debugView.
Le jeu ne freeze pas sur la console (testé avec la Satiator).

**Comment by BenjaminSiskoo** [comment]
D'après @vbt il semble que le soucis venait de ce bout de code :     if(chunk->alen>0 && chunk->alen <100000)
    for(i=0;i<4;i++)
    {
        if(!slPCMStat(&m_dat[i]))
        {
            //slPCMOff(&m_dat[i]);
            //slPCMParmChange(&m_dat[i]);
            slSndFlush() ;
// vbt 26/07/2020 : ? remettre    
                m_dat[i].mode= _PCM8Bit;
            slPCMOn(&m_dat[i],chunk->abuf,chunk->alen);
                break;
        }         
    }

**Comment by BenjaminSiskoo** [comment]
Est ce utile de la garder @fafling ?

**Comment by fafling** [comment]
Oui, même si la dernière WIP ne freeze plus aussi rapidement, mais il suffit de jouer un peu pour reproduire alors que ce n'est pas le cas sur la console.
